src/ispc.h

```diff
@@ -426,9 +429,12 @@ class Target {
         is 32 on SSE/AVX, since that matches the HW better. */
     int m_maskBitCount;
 
-    /** Indicates whether the target has native support for float/half
-        conversions. */
-    bool m_hasHalf;
+    /** Indicates whether the target has native support for float/half conversions. */
+    bool m_hasHalfConverts;
+
+    /** Indicates whether the target has full native support for float16 type, i.e.
+        arithmetic operations, rsqrt, rcp, etc. */
+    bool m_hasHalfFullSupport;
```

Comment:
What is difference between `m_hasHalfFullSupport` and `m_hasFp16Support` that we already have?

Reply:
That's very good sighting! I've added a TODO to add merge them. Right now they are different entities because m_hasHalfFullSupport is not enabled on ARM, as we need to differentiate between platform that support and do not support fp16 on ARM